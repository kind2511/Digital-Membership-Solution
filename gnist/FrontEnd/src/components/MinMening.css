body, html {
  margin: 0;
  padding: 0;
  overflow-x: hidden; 
}

.minMening-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;  
  padding: 20px 0;
  height: 80vh;  
  max-height: 80vh;  
}

.questions-container {
  width: 90%; 
  max-height: calc(80vh - 200px);  
  overflow-y: auto;  
  margin-bottom: 50px;  
  padding: 10px;
  border-radius: 5px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.question-block {
  margin-bottom: 20px;
  padding: 15px;
  background-color: #f7f7f7; 
  border-radius: 5px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
}

.question-title {
  margin-bottom: 10px;
  color: #333; 
  font-size: 1.2rem; 
}

.answers-list {
  list-style-type: none; 
  padding: 0;
}

.answer {
  background-color: #fff; 
  margin-bottom: 5px;
  padding: 8px 15px;
  border-radius: 5px;
  box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.05); 
  transition: background-color 0.3s; 
}

.answer:hover {
  background-color: #e9ecef; 
}

.divider {
  width: 90%;
  border-top: 1px solid #D9E3F0;
  margin-top: 20px; 
}

.title {
  text-align: center;
  font-size: 18px;
  margin-bottom: 20px;
}

.forslag-form {
  display: flex;
  flex-direction: column;
  width: 80%;
  max-width: 500px;
  margin-top: 1vh;
}

.forslag-input, .forslag-textarea {
  margin-bottom: 20px;
  padding: 10px;
  border-radius: 5px;
  border: 1px solid #ccc;
}

.forslag-textarea {
  height: 100px;
  resize: vertical;
}

.forslag-submit {
  background-color: #005A9E;
  color: #FFFFFF;
  border: none;
  padding: 10px 20px;
  margin: 10px;
  cursor: pointer;
  font-weight: bold;
  border-radius: 5px;
}

.forslag-submit:hover {
  background-color: #004080;
  color: #FFFFFF;
}

.forslag-modal-overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0,0,0,0.5);
  z-index: 2000; 
}

.forslag-modal-content {
  background-color:  #FFFFFF;
  padding: 20px;
  border-radius: 20px;
  box-shadow: 0 4px 8px rgba(0,0,0,0.2);
  text-align: center;
  font-weight: bold;
  width: 300px;
  border: 1px solid #D9E3F0;
}

.forslag-modal-content button {
  background-color: #005A9E;
  color:  #FFFFFF;
  border: none;
  padding: 10px 20px;
  margin: 10px;
  cursor: pointer;
  font-weight: bold;
  border-radius: 4px;
}

.forslag-modal-content button:hover {
  background-color: #004080;  
  color: #FFFFFF;
}

.confirmation-message {
  padding: 5px;
  margin-top: 10px;
  background-color: #52C41A;  
  color:  #FFFFFF;
  border-radius: 5px;
  display: inline-block;  
  animation: fadein 0.5s ease-in-out; 
  font-size: 13px;
}

@keyframes fadein {
  from { opacity: 0; }
  to   { opacity: 1; }
}

.cancel-button {
  background-color: red;
  color: white;

}

.answer-button {
  background-color: green;
  color: white;
}

.button-group {
  display: flex;
  justify-content: space-around;
  margin-top: 10px;
}
