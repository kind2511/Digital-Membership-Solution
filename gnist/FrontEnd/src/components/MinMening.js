import React, { useState, useEffect } from "react";
import './MinMening.css';

function MinMening() {
  const [title, setTitle] = useState('');
  const [description, setDescription] = useState('');
  const [questions, setQuestions] = useState([]);

  useEffect(() => {
    const fetchQuestions = async () => {
      try {
        const response = await fetch('http://127.0.0.1:8000/digital_medlemsordning/get_all_questions');
        const data = await response.json();
        setQuestions(data.questions); 
      } catch (error) {
        console.error("Error fetching questions:", error);
      }
    };

    fetchQuestions();
  }, []);

  const handleSubmit = async (event) => {
    event.preventDefault();
    const requestBody = {
      title: title,
      description: description
    };

    try {
      const response = await fetch('http://127.0.0.1:8000/digital_medlemsordning/create_suggestion/', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify(requestBody)
      });

      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }
      const data = await response.json();
      console.log('Suggestion sent successfully:', data);
      setTitle('');
      setDescription('');

    } catch (error) {
      console.error("Error sending suggestion:", error);
    }
  };

  return (
    <div className="minMening-container">
      <div className="questions-container">
        {questions.map((question) => (
          <div key={question.questionID} className="question-block">
            <h2 className="question-title">{question.question}</h2>
            <ul className="answers-list">
              {question.answers.map((answer) => (
                <li key={answer.answer_id} className="answer">
                  {answer.answer_text}
                </li>
              ))}
            </ul>
          </div>
        ))}
      </div>
      <hr className="divider" />
      <h1 className="title">Send Inn Forslag</h1>
      <form className="forslag-form" onSubmit={handleSubmit}>
        <input
          type="text"
          className="forslag-input"
          placeholder="Tittel"
          value={title}
          onChange={(e) => setTitle(e.target.value)}
          required
        />
        <textarea
          className="forslag-textarea"
          placeholder="Beskrivelse"
          value={description}
          onChange={(e) => setDescription(e.target.value)}
          required
        />
        <button type="submit" className="forslag-submit">Send</button>
      </form>
    </div>
  );
}

export default MinMening;
