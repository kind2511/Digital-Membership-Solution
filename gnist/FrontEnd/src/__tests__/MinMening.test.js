import React from 'react';
import renderer from 'react-test-renderer';
import MinMening from '../components/MinMening';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { Auth0Context } from '@auth0/auth0-react';

// Mock axios for the test
const mock = new MockAdapter(axios);
const questionsMock = [
  { questionID: 1, question: 'What is your favorite color?', answers: [{ answer_id: 1, answer_text: 'Red' }, { answer_id: 2, answer_text: 'Blue' }] },
  { questionID: 2, question: 'What is your favorite animal?', answers: [{ answer_id: 3, answer_text: 'Dog' }, { answer_id: 4, answer_text: 'Cat' }] },
];

mock.onGet('http://localhost:8000/digital_medlemsordning/get_all_questions').reply(200, { questions: questionsMock });
mock.onPost('http://localhost:8000/digital_medlemsordning/create_suggestion/').reply(201);
mock.onPost(/http:\/\/localhost:8000\/digital_medlemsordning\/submit_response\/.*/).reply(201);

// Mock Auth0 context
const mockAuth0Context = {
  getAccessTokenSilently: jest.fn().mockResolvedValue('mocked_token'),
  user: { sub: 'auth0|123456789' },
};

test('MinMening component snapshot', () => {
  const component = renderer.create(
    <Auth0Context.Provider value={mockAuth0Context}>
      <MinMening />
    </Auth0Context.Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
