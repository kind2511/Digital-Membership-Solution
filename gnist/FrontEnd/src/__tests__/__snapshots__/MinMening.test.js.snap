// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinMening component snapshot 1`] = `
<div
  className="minMeningUniqueContainer"
>
  <section
    className="uniqueSuggestionSection"
  >
    <h2>
      Send Inn Forslag
    </h2>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Tittel"
        required={true}
        type="text"
        value=""
      />
      <textarea
        onChange={[Function]}
        placeholder="Ditt forslag"
        required={true}
        value=""
      />
      <button
        className="uniqueSendInnButton"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Send Inn
      </button>
    </form>
  </section>
  <section
    className="uniqueQuestionSection"
  >
    <h2>
      Avstemninger
    </h2>
  </section>
</div>
`;
